src/visualization/types.jl

```diff
@@ -330,6 +330,44 @@ function PlotData2D(u::StructArray, mesh, equations, dg::DGMulti, cache;
   return PlotData2DTriangulated(x_plot, y_plot, u_plot, t, x_face, y_face, face_data, variable_names)
 end
 
+
+function PlotData2D(u_ode::AbstractVector, semi::SemidiscretizationCoupled;
+                    solution_variables=nothing, nvisnodes=2*polydeg(semi), kwargs...)
+  @assert ndims(semi) == 2 "unsupported number of dimensions $ndims (must be 2)"
+
+  x_vec = []
+  y_vec = []
+  data_vec = []
+  pd = 0
+
+  @unpack semis, u_indices = semi
+  for i in 1:nmeshes(semi)
+    mesh, equations, solver, cache = mesh_equations_solver_cache(semis[i])
+
+    u = wrap_array(u_ode[u_indices[i]], semis[i])
+
+    pd = PlotData2D(u, mesh::Union{StructuredMesh, UnstructuredMesh2D, P4estMesh}, equations, solver, cache;
+                    solution_variables=solution_variables, nvisnodes=nvisnodes)
```

Comment:
```suggestion
    pd = PlotData2D(u, mesh, equations, solver, cache;
                    solution_variables=solution_variables, nvisnodes=nvisnodes)
```
If I am not mistaken, this is a copy paste error. Btw, can you update the plot in the PR description to show what the visualization looks like right now?

Reply:
I'm not sure this is copy and paste error since `TreeMesh` is not included in this union. IIRC there's a `PlotData2D` specialized to `TreeMesh` for Cartesian-type plotting.